### PR TITLE
fix: Update PeerDid resolver to support the multiple service individually encoded

### PR DESCRIPTION
**Pull request by mineme0110**
The peer-did-jvm library currently does not correctly resolve peer DIDs that have multiple individually encoded services, as outlined in the peer DID specification. For instance, it cannot accurately handle DIDs similar to the example provided. 

did:peer:2.Vz6Mkj3PUd1WjvaDhNZhhhXQdz5UnZXmS7ehtx8bsPpD47kKc.Ez6LSg8zQom395jKLrGiBNruB9MM6V8PWuf2FpEy4uRFiqQBR.SeyJ0IjoiZG0iLCJzIjp7InVyaSI6Imh0dHA6Ly9leGFtcGxlLmNvbS9kaWRjb21tIiwiYSI6WyJkaWRjb21tL3YyIl0sInIiOlsiZGlkOmV4YW1wbGU6MTIzNDU2Nzg5YWJjZGVmZ2hpI2tleS0xIl19fQ.SeyJ0IjoiZG0iLCJzIjp7InVyaSI6Imh0dHA6Ly9leGFtcGxlLmNvbS9hbm90aGVyIiwiYSI6WyJkaWRjb21tL3YyIl0sInIiOlsiZGlkOmV4YW1wbGU6MTIzNDU2Nzg5YWJjZGVmZ2hpI2tleS0yIl19fQ

To address this issue, This PR updates the regex validation to support multiple services encoded separately. Additionally, it modifies the peer DID resolver to ensure correct resolution of such DIDs. 
This change aligns with the recent updates in the peer DID method specification, as seen in PR #67 on the  repository  
https://github.com/decentralized-identity/peer-did-method-spec/pull/67

These improvements are designed to enhance the library's functionality without breaking any existing behavior of the library. 

However, in future, I will raise a few more small PRs to align the library with the latest specifications.
https://identity.foundation/peer-did-method-spec

### CODE REVIEW

**Comment by mineme0110** [comment]
@andkononykhin @dhh1128  @ashcherbakov  
Hello Guys can you please review this PR

**Review by dhh1128 — APPROVED** [review]
Looks good to me.

**Comment by dhh1128** [comment]
I have left a positive review, but I do not have write access to this repo anymore, so I can't merge it. If you can't get the attention of maintainers, you may need to fork it and do the fix there.

**Comment by ashcherbakov** [comment]
I approved the PR, but I also don't have write access permissions anymore.

**Comment by mineme0110** [comment]
> I have left a positive review, but I do not have write access to this repo anymore, so I can't merge it. If you can't get the attention of maintainers, you may need to fork it and do the fix there.

Thank You I will try to find out who is the maintainer

**Comment by mineme0110** [comment]
@dkulic 
Hello  Please can you review this PR 
Thank You

**Review by dkulic — APPROVED** [review]

**Comment by dkulic** [comment]
Thanks @mineme0110 for this contribution.

**Comment by mineme0110** [comment]
@dkulic Thank you 
Will you be able to do a release, I don't have the necessary permissions 
https://github.com/sicpa-dlab/peer-did-jvm/blob/main/docs/release.md

**Comment by mineme0110** [comment]
@andkononykhin  Please will you you be able to do a release, I don't have the necessary permissions
https://github.com/sicpa-dlab/peer-did-jvm/blob/main/docs/release.md

**Comment by mineme0110** [comment]
@andkononykhin 
Reminder againe Please will you be able to do a release

**Comment by dkulic** [comment]
@mineme0110 I released 0.4.0.

**Pull request merged**
